pkg/operator/baremetal_config.go

```diff
@@ -22,68 +24,107 @@ const (
 	baremetalKernelUrlSubPath      = "images/ironic-python-agent.kernel"
 	baremetalRamdiskUrlSubPath     = "images/ironic-python-agent.initramfs"
 	baremetalIronicEndpointSubpath = "v1/"
+	provisioningNetworkManaged     = "Managed"
+	provisioningNetworkUnmanaged   = "Unmanaged"
+	provisioningNetworkDisabled    = "Disabled"
 )
 
 // Provisioning Config needed to deploy Metal3 pod
 type BaremetalProvisioningConfig struct {
 	ProvisioningInterface     string
 	ProvisioningIp            string
 	ProvisioningNetworkCIDR   string
-	ProvisioningDHCPExternal  bool
 	ProvisioningDHCPRange     string
 	ProvisioningOSDownloadURL string
+	ProvisioningNetwork       string
 }
 
 func getBaremetalProvisioningConfig(dc dynamic.Interface, configName string) (BaremetalProvisioningConfig, error) {
 	provisioningClient := dc.Resource(provisioningGVR)
 	provisioningConfig, err := provisioningClient.Get(context.Background(), configName, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		glog.V(3).Infof("Baremetal provisioning CR %s is not found", configName)
+		return BaremetalProvisioningConfig{}, nil
+	}
 	if err != nil {
-		glog.Errorf("Error getting config from Baremetal provisioning CR %s", configName)
-		return BaremetalProvisioningConfig{}, err
+		errMsg := fmt.Sprintf("Error getting config from Baremetal provisioning CR %s", configName)
+		return BaremetalProvisioningConfig{}, errors.Wrap(err, errMsg)
```

Comment:
We prefer to use the go standard error wrapping rather than the errors package, could you update your error wraps to use `fmt.Errorf` with the `%w` format instead? Out of interest, are you going to be unwrapping these errors at any point later? If not, it may not be worth wrapping at all in which case you can you just use `%v` instead of `%w`
```suggestion
		return BaremetalProvisioningConfig{}, fmt.Errorf("Error getting config from Baremetal provisioning CR %s: %w", configName, err)
```